airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -410,14 +411,18 @@ protected long getActualCursorRecordCount(final Connection connection,
 
   @Override
   public JdbcDatabase createDatabase(final JsonNode sourceConfig) throws SQLException {
+    return createDatabase(sourceConfig, JdbcDataSourceUtils::getConnectionProperties);
```

Comment:
Let's just override the entire createDatabase method in MySqlSource to support this use case, with a comment on why it needs overriding